src/chocolatey/infrastructure.app/services/NugetService.cs

```diff
@@ -793,6 +799,23 @@ Version was specified as '{0}'. It is possible that version
 
                 foreach (SourcePackageDependencyInfo packageDependencyInfo in resolvedPackages)
                 {
+                    // Don't attempt to action this package if dependencies failed.
+                    if (packageDependencyInfo != null && packageResultsToReturn.Any(r => r.Value.Success != true && packageDependencyInfo.Dependencies.Any(d => d.Id == r.Value.Identity.Id)))
+                    {
+                        var logMessage = StringResources.ErrorMessages.DependencyFailedToInstall.FormatWith(packageDependencyInfo.Id);
+                        var x = new PackageResult(packageDependencyInfo.Id, packageDependencyInfo.Version.ToStringSafe(), string.Empty);
+                        var nullResult = packageResultsToReturn.GetOrAdd(packageDependencyInfo.Id, x);

```

Comment:
Not a huge fan of `nullResult` either.  Perhaps some additional comments to say what is going on here?

Reply:
It looks like we didn't even need the `nullResult`. I've updated all the instances and left a comment on the one place where we needed it.